src/components/dialogs/element-creation-dialog.tsx

```diff
@@ -0,0 +1,226 @@
+
+    const clear = useCallback(() => {
+        reset(emptyFormData);
+    }, [reset]);
+
+    const fetchDefaultDirectoryForStudy = useCallback(() => {
+        // @ts-expect-error TODO: manage null case
+        fetchDirectoryElementPath(studyUuid).then((res) => {
+            if (!res || res.length < 2) {
+                snackError({
+                    messageTxt: 'unknown study directory',
+                    headerId: 'studyDirectoryFetchingError',
+                });
+                return;
+            }
+            const parentFolderIndex = res.length - 2;
+            const { elementUuid, elementName } = res[parentFolderIndex];
+            setDestinationFolder({
+                id: elementUuid,
+                name: elementName,
+            });
+        });
+    }, [studyUuid, snackError]);
+
+    useEffect(() => {
+        if (prefixIdForGeneratedName) {
+            const getCurrentDateTime = () => new Date().toISOString();
+            const formattedMessage = intl.formatMessage({
+                id: prefixIdForGeneratedName,
+            });
+            const dateTime = getCurrentDateTime();
+            reset(
+                {
+                    ...emptyFormData,
+                    [NAME]: `${formattedMessage}-${dateTime}`,
+                },
+                { keepDefaultValues: true }
+            );
+        }
+    }, [prefixIdForGeneratedName, intl, reset]);
+
+    useEffect(() => {
+        if (open && studyUuid) {
+            fetchDefaultDirectoryForStudy();
+        }
+    }, [fetchDefaultDirectoryForStudy, studyUuid, open]);
+
+    const handleChangeFolder = () => {
+        setDirectorySelectorOpen(true);
+    };
+
+    const setSelectedFolder = (folder: TreeViewFinderNodeProps[]) => {
+        if (folder?.length > 0 && folder[0].id !== destinationFolder?.id) {
+            const { id, name } = folder[0];
+            setDestinationFolder({ id, name });
+        }
+        setDirectorySelectorOpen(false);
+    };
+
+    const handleSave = useCallback(
+        (values: FormData) => {
+            if (destinationFolder) {
+                const creationData: IElementCreationDialog = {
+                    ...values,
+                    [FOLDER_NAME]: destinationFolder.name,
+                    [FOLDER_ID]: destinationFolder.id as UUID,
+                };
+                onSave(creationData);
+            }
+        },
+        [onSave, destinationFolder]
+    );
+
+    const folderChooser = (
+        <Grid container item paddingTop={1}>
+            <Grid item>
+                <Button onClick={handleChangeFolder} variant="contained" size={'small'}>
+                    <FormattedMessage id={'showSelectDirectoryDialog'} />
+                </Button>
+            </Grid>
+            <Typography m={1} component="span">
+                <Box fontWeight={'fontWeightBold'}>
+                    {destinationFolder == null ? <CircularProgress /> : destinationFolder.name}
```

Comment:
i don't see a case where destinationFolder could be null ?

Reply:
it cant.
I will remove '==null' (and invert the return data)